p2p/mdns/mdns.go

```diff
@@ -0,0 +1,216 @@
+package mdns
```

Comment:
I was actually thinking discovery/mdns to better organize the packages. But this also works.

Reply:
Yes, that's better. Let's do that.